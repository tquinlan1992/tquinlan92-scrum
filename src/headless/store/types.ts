import { Ticket } from "@headless/database/PouchWrapper";
import { RouterState } from "react-router-redux";
import { ThunkAction } from "redux-thunk";
import { AnyAction } from "redux";

export type Tickets = Ticket[];

export interface ClosedTicket extends Ticket {
    closed: true;
    sprint: false;
} 

export interface SprintTicket extends Ticket {
    sprint: true;
}

export interface BacklogTicket extends Ticket {
    sprint: false;
    closed: false;
}

export type LoadingState = {
    value: boolean
};

export interface AddTicketState {
    storyPoint: null | number;
    description: string;
    title: string;
    open: boolean;
    editId: string;
}

export interface TicketListState {
    tickets: Ticket[];
    sprintTickets: SprintTicket[];
    closedTickets: ClosedTicket[];
    backlogTickets: BacklogTicket[];
    showCloseSprintDialog: boolean;
}

export interface CloseSprintDialogState {
    sprintName: string;
}

export interface AddTicketState {
    storyPoint: null | number;
    description: string;
    title: string;
}


export interface AppState {
    routing: RouterState;
    ticketList: TicketListState;
    addTicket: AddTicketState;
    loading: LoadingState;
    closeSprintDialog: CloseSprintDialogState;
}

export type AppThunkAction = ThunkAction<void, AppState, void, AnyAction>;
