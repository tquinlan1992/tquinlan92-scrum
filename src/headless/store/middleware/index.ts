// import saveToLocalStorageState from './saveToLocalStorageState';
import thunk from 'redux-thunk';
import { routerÂ } from './router';
import logger from 'redux-logger';

export const middleware = [
    // saveToLocalStorageState,
    router,
    thunk,
    logger
];
