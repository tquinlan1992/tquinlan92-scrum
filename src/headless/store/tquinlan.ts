import { Dictionary, mapValues, omit, merge } from "lodash";
import { Reducer, AnyAction } from "redux";
import { makeSimpleReducer, ActionCreator } from "tquinlan92-typescript-redux-utils";

export function getActions<T extends { [key: string]: {actions?: Dictionary<any>} }>(creators: T): { [P in keyof T]: T[P]['actions'] } {
    return mapValues(creators, "actions") as { [P in keyof T]: T[P]['actions'] };
} 

export function makeNestedSimpleReducerSimpleActions<AppState>(state: any) {
    const actionsReducers = mapValues(state, (value, key) => {
        return makeSimpleReducer(key, omit(value, 'actions'));
    });//{ [P in keyof T]: T[P]['actionCreator'] 
    const reducers = mapValues(actionsReducers, 'reducer');
    const actions = getActions(actionsReducers);

    return {
        actions,
        reducers
    } as {
        reducers: {
            [P in keyof AppState]: Reducer<AppState[P], AnyAction>
        };
        actions: {
            [P in keyof AppState]: {
                [A in keyof AppState[P]]: ActionCreator<AppState[P][A]>
            } & {
                reset: ActionCreator<null>;
                setAll: ActionCreator<AppState[P]>;
                set: ActionCreator<Partial<AppState[P]>>;
            } & {
                [A in keyof AppState[P]]: AppState[P][A]
            }
        };
    };
}

export function makeNestedSimpleStore<State, ThunkActions>(state: State, thunkActions?: ThunkActions) {
    const { actions: actions1, reducers: reducers2 } = makeNestedSimpleReducerSimpleActions<State>(state);
    const actionsWithThunks = merge(actions1, thunkActions);
    return {
        actions: actionsWithThunks,
        reducers: reducers2
    }
}