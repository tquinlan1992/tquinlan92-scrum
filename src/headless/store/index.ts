import { createStore, applyMiddleware, AnyAction, Store, combineReducers } from "redux";
import { middleware } from './middleware';
import { composeWithDevTools } from 'redux-devtools-extension';
import { routerReducer, RouterState } from "react-router-redux";
import { TicketListState } from '@components/TicketList/redux';
import { AddTicketState, addTicketDialogReducer } from '@components/AddTicketDialog/redux';
import { LoadingState, loadingReducer } from '@components/Loading/redux';
import { AppState } from '../store';
import { ticketListReducer } from '../../components/TicketList/redux';
import { ThunkAction } from "redux-thunk";
import { CloseSprintDialogState, closeSprintDialogReducer } from '@components/CloseSprintDialog/redux';


const appReducer = combineReducers<AppState>({
    routing: routerReducer,
    addTicket: addTicketDialogReducer,
    ticketList: ticketListReducer,
    loading: loadingReducer,
    closeSprintDialog: closeSprintDialogReducer
});

declare module 'redux' {
    export type GenericStoreEnhancer = any;
}

export type AppThunkAction = ThunkAction<void, AppState, void, AnyAction>;

export interface Doc {
    edit: string;
}

export interface VoiceToCode {
    text: string;
}

export interface AddTicketState {
    storyPoint: null | number;
    description: string;
    title: string;
}

export interface AppState {
    routing: RouterState;
    ticketList: TicketListState;
    addTicket: AddTicketState;
    loading: LoadingState;
    closeSprintDialog: CloseSprintDialogState;
}

const reduxStore: Store<AppState, AnyAction> = createStore<AppState, AnyAction, void, void>(appReducer, {}, composeWithDevTools<any, {}>(applyMiddleware(...middleware)));

export { reduxStore };
