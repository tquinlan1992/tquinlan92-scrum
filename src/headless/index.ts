import { reduxStore } from './store';

export { reduxStore };
