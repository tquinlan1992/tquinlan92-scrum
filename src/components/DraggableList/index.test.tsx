import { DraggableList } from './';
import { shallow } from 'enzyme';
import * as React from 'react';
import { AppState } from '../../headless/store';

describe('test snapshot of component', () => {
    it('should render', () => {
        const props = {
            listItems: [
                {
                _id: '_id1',
                title: 'title1'
            },
            {
                _id: '_id2',
                title: 'title2'
            }
        ],
            updateItems: jest.fn(),
            title: 'title'
        };
        const result = shallow(<DraggableList {...props} />);
        expect(result).toMatchSnapshot();
    });
});
