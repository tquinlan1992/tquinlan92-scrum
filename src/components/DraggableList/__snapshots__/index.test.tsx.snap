// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot of component should render 1`] = `
<Fragment>
  <WithStyles(Typography)
    gutterBottom={true}
    variant="h6"
  >
    title
  </WithStyles(Typography)>
  <DragDropContext
    onDragEnd={[Function]}
  >
    <Connect(Droppable)
      direction="vertical"
      droppableId="droppable"
      ignoreContainerClipping={false}
      isCombineEnabled={false}
      isDropDisabled={false}
      type="DEFAULT"
    >
      <Component />
    </Connect(Droppable)>
  </DragDropContext>
</Fragment>
`;
